src/components/Table/GridTable.tsx

```diff
@@ -277,15 +278,33 @@ export function GridTable<R extends Kinded, X extends Only<GridTableXss, X> = an
     return rows;
   }, [columns, rows, sortOn, sortState, caseSensitive]);
 
+  const unmatchedSelectedDataRows = useComputed(
+    () => tableState.unmatchedSelectedRows as GridDataRow<R>[],
+    [tableState],
+  );
+  // Sort the `unmatchedSelectedDataRows` separately
```

Comment:
Just curious, but why does the sort need to be separate? Maybe work that into the in-source comment, i.e. `// Sort ... separately b/c ...rationale...`

Reply:
Sure can do.
The reasoning is because the current sorting logic sorts within the groups and these "unmatched" rows are now displayed in a flat list.